pkg/distro/rhel8/partition_tables.go

```diff
@@ -423,3 +247,188 @@ var edgeBasePartitionTables = distro.BasePartitionTableMap{
+	if osVersion == "8.9" {
+		// 8.9: has 500 MiB /boot
+		return distro.BasePartitionTableMap{
+			arch.ARCH_X86_64.String(): disk.PartitionTable{
+				UUID: "D209C89E-EA5E-4FBD-B161-B461CCE297E0",
+				Type: "gpt",
+				Partitions: []disk.Partition{
+					{
+						Size:     1 * common.MebiByte,
+						Bootable: true,
+						Type:     disk.BIOSBootPartitionGUID,
+						UUID:     disk.BIOSBootPartitionUUID,
+					},
+					{
+						Size: 200 * common.MebiByte,
+						Type: disk.EFISystemPartitionGUID,
+						UUID: disk.EFISystemPartitionUUID,
+						Payload: &disk.Filesystem{
+							Type:         "vfat",
+							UUID:         disk.EFIFilesystemUUID,
+							Mountpoint:   "/boot/efi",
+							Label:        "EFI-SYSTEM",
+							FSTabOptions: "defaults,uid=0,gid=0,umask=077,shortname=winnt",
+							FSTabFreq:    0,
+							FSTabPassNo:  2,
+						},
+					},
+					{
+						Size: 500 * common.MebiByte,
+						Type: disk.XBootLDRPartitionGUID,
+						UUID: disk.FilesystemDataUUID,
+						Payload: &disk.Filesystem{
+							Type:         "xfs",
+							Mountpoint:   "/boot",
+							Label:        "boot",
+							FSTabOptions: "defaults",
+							FSTabFreq:    0,
+							FSTabPassNo:  0,
+						},
+					},
+					{
+						Size: 2 * common.GibiByte,
+						Type: disk.FilesystemDataGUID,
+						UUID: disk.RootPartitionUUID,
+						Payload: &disk.Filesystem{
+							Type:         "xfs",
+							Label:        "root",
+							Mountpoint:   "/",
+							FSTabOptions: "defaults",
+							FSTabFreq:    0,
+							FSTabPassNo:  0,
+						},
+					},
+				},
+			},
+			arch.ARCH_AARCH64.String(): disk.PartitionTable{
+				UUID: "D209C89E-EA5E-4FBD-B161-B461CCE297E0",
+				Type: "gpt",
+				Partitions: []disk.Partition{
+					{
+						Size: 200 * common.MebiByte,
+						Type: disk.EFISystemPartitionGUID,
+						UUID: disk.EFISystemPartitionUUID,
+						Payload: &disk.Filesystem{
+							Type:         "vfat",
+							UUID:         disk.EFIFilesystemUUID,
+							Mountpoint:   "/boot/efi",
+							Label:        "EFI-SYSTEM",
+							FSTabOptions: "defaults,uid=0,gid=0,umask=077,shortname=winnt",
+							FSTabFreq:    0,
+							FSTabPassNo:  2,
+						},
+					},
+					{
+						Size: 500 * common.MebiByte,
+						Type: disk.XBootLDRPartitionGUID,
+						UUID: disk.FilesystemDataUUID,
+						Payload: &disk.Filesystem{
+							Type:         "xfs",
+							Mountpoint:   "/boot",
+							Label:        "boot",
+							FSTabOptions: "defaults",
+							FSTabFreq:    0,
+							FSTabPassNo:  0,
+						},
+					},
+					{
+						Size: 2 * common.GibiByte,
+						Type: disk.FilesystemDataGUID,
+						UUID: disk.RootPartitionUUID,
+						Payload: &disk.Filesystem{
+							Type:         "xfs",
+							Label:        "root",
+							Mountpoint:   "/",
+							FSTabOptions: "defaults",
+							FSTabFreq:    0,
+							FSTabPassNo:  0,
+						},
+					},
+				},
+			},
+		}
+	}
+
+	// All other RHEL 8 minor versions have /boot-less x86_64 layout.
```

Comment:
Thanks for the comment - now I'm curious - so 8.0 to 8.8 have a /boot-less layout, 8.9 as 500mb and 8.10 is /boot-less again? Or is it more complex and I'm just misreading this?

Reply:
See my latest comment, I simplified this in the end.